integration-tests/deployment/ccip/add_chain_test.go

```diff
@@ -0,0 +1,157 @@
+package ccipdeployment
+
+import (
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink/integration-tests/deployment"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/offramp"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/router"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+func TestAddChainInbound(t *testing.T) {
+	// 4 chains where the 4th is added after initial deployment.
+	e := NewEnvironmentWithCRAndJobs(t, logger.TestLogger(t), 4)
+	require.Equal(t, len(e.Nodes), 5)
+	state, err := LoadOnchainState(e.Env, e.Ab)
+	require.NoError(t, err)
+	// Take first non-home chain as the new chain.
+	newChain := e.Env.AllChainSelectorsExcluding([]uint64{e.HomeChainSel})[0]
+	// We deploy to the rest.
+	initialDeploy := e.Env.AllChainSelectorsExcluding([]uint64{newChain})
+
+	ab, err := DeployCCIPContracts(e.Env, DeployCCIPContractConfig{
+		HomeChainSel:     e.HomeChainSel,
+		ChainsToDeploy:   initialDeploy,
+		CCIPOnChainState: state,
+	})
+	require.NoError(t, err)
+	require.NoError(t, e.Ab.Merge(ab))
+	state, err = LoadOnchainState(e.Env, e.Ab)
+	require.NoError(t, err)
+
+	// Connect all the existing lanes.
+	for _, source := range initialDeploy {
+		for _, dest := range initialDeploy {
+			if source != dest {
+				require.NoError(t, AddLane(e.Env, state, source, dest))
+			}
+		}
+	}
+
+	//  Deploy contracts to new chain
+	newAddresses, err := DeployChainContracts(e.Env, e.Env.Chains[newChain], deployment.NewMemoryAddressBook())
+	require.NoError(t, err)
+	require.NoError(t, e.Ab.Merge(newAddresses))
+	state, err = LoadOnchainState(e.Env, e.Ab)
+	require.NoError(t, err)
+
+	// Transfer onramp/fq ownership to timelock.
+	// Enable the new dest on the test router.
+	for _, source := range initialDeploy {
```

Comment:
Semantically distinct code blocks below (already designated by comments) can probably be moved to functions to make this test read/understand better

Reply:
agree - was thinking we wait for another proposal to see if there are nice re-usable utilities we can extract